buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -974,19 +994,19 @@ private Path getExtractedDistributionDir() {
     }
 
     @Classpath
-    private List<File> getDistributionClasspath() {
-        ArrayList<File> files = new ArrayList<>(project.fileTree(getExtractedDistributionDir())
-            .matching(filter -> filter.include("**/*.jar"))
-            .getFiles());
-        files.sort(Comparator.comparing(File::getName));
-
-        return files;
+    private Iterable<File> getDistributionClasspath() {
+        FileTree jarFiles = distribution.getExtracted().getFileTree(project).matching(filter -> filter.include("**/*.jar"));
+        return () -> {
```

Comment:
This is very clever 👍. Thing is, it doesn't actually do what we want, lol. So my intuition was incorrect and returning an `Iterable` doesn't actually defer anything at task graph calculation time. That's because this iterable could be some other `Buildable` type, Gradle has to iterate this collection and add each entry as input. Due to type erasure we have no idea at runtime what this is a collection of, so we have to rip through it.

The interesting thing here though is that our assumption that calling `getSingleFile()` would cause resolution is actually incorrect. This doesn't actually trigger any work. It just returns a path to the empty directory as we are just resolving the configuration, but not it's _artifacts_. So all this effort to defer that was in vain 🤦‍♂ .

With that knowledge in hand we might be able to simplify some of this logic now that we knowing calling `getSingleFile()` on a configuration which includes a single directory artifact is fine.